tests/test_home.py

```diff
@@ -1132,3 +1133,52 @@ async def test_emergency_heat(aiohttp_session: aiohttp.ClientSession) -> None:
     assert zone.get_setpoint_status() == "Run Schedule - None"
     assert zone.is_calling() is True
     assert zone.is_in_permanent_hold() is False
+
+
+@pytest.mark.asyncio
+async def test_humidity_and_fan_mode(aiohttp_session: aiohttp.ClientSession, mock_aioresponse: aioresponses) -> None:
+    """Tests for preventing an API timeout when updating humidity
+    and fan modes to the same value
+    """
+    nexia = NexiaHome(aiohttp_session)
+    devices_json = json.loads(await load_fixture("mobile_house_issue_33758.json"))
+    nexia.update_from_json(devices_json)
+
+    thermostat: NexiaThermostat = nexia.get_thermostat_by_id(12345678)
+
+
+    # Test setting fan mode to the same value (should not trigger API call)
+    await thermostat.set_fan_mode("Auto")  # Attempt to set to the same value again
+    # Nothing should happen
+
+    # Mock the API call to prevent actual HTTP requests
+    mock_aioresponse.post(
+        "https://www.mynexia.com/mobile/xxl_thermostats/12345678/fan_mode",
+        payload={
+            # TODO: put a mock responses here
```

Comment:
I pushed a sample to get you started. You'll need to fill in the response from the post request